cmd/auth-rpc-client.go

```diff
@@ -80,13 +80,13 @@ func newAuthRPCClient(config authConfig) *AuthRPCClient {
 
 // Login - a jwt based authentication is performed with rpc server.
 func (authClient *AuthRPCClient) Login() (err error) {
-	authClient.Lock()
-	defer authClient.Unlock()
-
+	authClient.RLock()
 	// Return if already logged in.
 	if authClient.authToken != "" {
+		authClient.RUnlock()
 		return nil
 	}
+	authClient.RUnlock()
```

Comment:
It is to use right kind of locks of reading authToken() currently they block on each other. when using Lock() instead of RLock().  We can make login RPC inside to be blocking.. But reading shouldn't be blocking.